Sources/MessagingPush/Store/PushHistory.swift

```diff
@@ -0,0 +1,47 @@
+import CioInternalCommon
+import Foundation
+
+protocol PushHistory: AutoMockable {
+    func hasHandledPushClick(deliveryId: String) -> Bool
+    func handledPushClick(deliveryId: String)
+}
+
+// sourcery: InjectRegister = "PushHistory"
+class PushHistoryImpl: PushHistory {
```

Comment:
New class meant to prevent duplicate push metrics and duplicate deep link handling when a push is clicked on. 

We have encountered this problem for a while. This is meant to be a long-term solution to the problem by keeping track of last 100 delivery-ids the SDK has handled.